plugins/filter-highlight-all/prism-filter-highlight-all.js

```diff
@@ -0,0 +1,106 @@
+(function () {
+
+	if (typeof self !== 'undefined' && !self.Prism) {
+		return;
+	}
+
+	// https://developer.mozilla.org/en-US/docs/Web/API/Element/matches#Polyfill
+	if (!Element.prototype.matches) {
+		Element.prototype.matches = Element.prototype.msMatchesSelector || Element.prototype.webkitMatchesSelector;
+	}
+
+	var script = Prism.util.currentScript();
+
+
+	/**
+	 * @type {Array<(element: HTMLElement) => boolean>}
+	 */
+	var filters = [];
+
+	var config = Prism.plugins.filterHighlightAll = {
+
+		/**
+		 * Filters the elements of `highlightAll` and `highlightAllUnder` such that only element for which the given
+		 * function returns `true` will be highlighted.
+		 *
+		 * @param {(value: { element: HTMLElement, language: string }) => boolean} condition
+		 */
+		filter: function (condition) {
```

Comment:
`addFilter`?

Reply:
I think it's enough to know that this function adds a filter. If the name should also reflect this behaviour, then we'd have to rename the selector functions as well.

Reply:
I'd almost go the other way and suggest then we should make all of them `addX`. I could also conceivably see `Prism.plugins.filterHighlightAll.add.{filter,selector,etc.}` as the API, making the `add` a namespace object, but that might be a bit...much.

Reply:
The naming problem I see with the namespace idea is that current verbs will become nouns, e.g. `filterSelector` -> `add.selectorFilter`. This is fine for "filter" but for "reject"... `rejectSelector` -> `add.selectorRejecter` ??

Also, I think it's enough to just say that these functions add a new filter.

Reply:
I was thinking `addFilterSelector` and `addRejectSelector` for both of those, as we're adding a "reject selector" (a selector that rejects). If I'm being pedantic, the function it creates is the "rejector".

My overall issue with the naming here is I'd prefer these method names not match array methods. The `add` prefix indicates that we're adding a thing that the plugin is holding onto and running later, rather than running the filter immediately.

Reply:
How about `add` (current `filter`), `addSelector`, `reject.add` (new), `reject.addSelector` ?
"Filter" is in the plugin's name so maybe we don't have to repeat it?

I would still keep the current `data-filter-selector` and `data-reject-selector` attributes.

Reply:
This works for me!